src/NHibernate/Id/SelectGenerator.cs

```diff
@@ -72,19 +81,42 @@ public partial class SelectGeneratorDelegate : AbstractSelectingDelegate
 			private readonly SqlString idSelectString;
 			private readonly IType idType;
 			private readonly IPostInsertIdentityPersister persister;
+			private readonly IEntityPersister entityPersister;
 
-			private readonly string uniqueKeyPropertyName;
-			private readonly IType uniqueKeyType;
+			private readonly string[] uniqueKeyPropertyNames;
+			private readonly IType[] uniqueKeyTypes;
 
 			internal SelectGeneratorDelegate(IPostInsertIdentityPersister persister, ISessionFactoryImplementor factory,
-			                                 string suppliedUniqueKeyPropertyName) : base(persister)
+			                                 string suppliedUniqueKeyPropertyNames) : base(persister)
 			{
 				this.persister = persister;
 				this.factory = factory;
-				uniqueKeyPropertyName = DetermineNameOfPropertyToUse((IEntityPersister) persister, suppliedUniqueKeyPropertyName);
 
-				idSelectString = persister.GetSelectByUniqueKeyString(uniqueKeyPropertyName);
-				uniqueKeyType = ((IEntityPersister) persister).GetPropertyType(uniqueKeyPropertyName);
+				if (persister is IEntityPersister entityPersister)
+				{
+					this.entityPersister = entityPersister;
+					uniqueKeyPropertyNames = DetermineNameOfPropertiesToUse(entityPersister, suppliedUniqueKeyPropertyNames);
+
+					uniqueKeyTypes = uniqueKeyPropertyNames.Select(p => entityPersister.GetPropertyType(p)).ToArray();
+				}
+				else if (persister is IQueryableCollection collectionPersister)
+				{
+					uniqueKeyPropertyNames = new string[2];
+					uniqueKeyTypes = new IType[2];
```

Comment:
Are the `uniqueKeyPropertyNames` and `uniqueKeyTypes` set in this path used at all?